src/main/java/uk/gov/pay/api/model/SelfLinks.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.pay.api.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import static javax.ws.rs.HttpMethod.GET;
+
+@ApiModel(value = "selfLinks", description = "Resource self link of a Payment")
+public class SelfLinks {
```

Comment:
I would rename this to simply `Links` and anything more complex could you the _Decorator pattern_ and we could call it `ExtendedLinks`, instead of calling `SelfAndNextLinks`, as any other change of implementation would require a change of name as well (again we are being a bit more loose with _Cohesion_ and _Single Responsibility_ principles) but covering our selves for future tech debt and I believe better code readability.